README.md

```diff
@@ -1,19 +1,90 @@
+  See Back-end > Building above.
+
+### Building
+
+This example plugin uses [Webpack](https://webpack.js.org/) for building the final JavaScript. Whatever build system you choose to use, the final result *MUST* adhere to the following rules:
+
+* 1 entry file *per extension page*.
+* The name of each entry file must correspond to the `page_id` of the registered page (see `src/main/java/org/sonarsource/plugins/example/web/MyPluginPageDefinition.java` and compare with the entry points in `conf/webpack/webpack.config.js`).
+* Each entry file must be located in the resulting JAR's `static/` folder.
+
+The building process should be included in your full packaging process. In this example plugin, `mvn package` will call `npm run build` prior to finalizing the JAR package.
+
+### Testing
+
+This project uses [Jest](https://jestjs.io/) for testing. Running `npm test` will run Jest in `--watch` mode. You can find the configuration for Jest in `package.json`.
+
+### How to use these files
+
+It is recommended you check out the sources in `src/main/js/` directly. The code is well commented, and provides real-world examples on how to interact with SonarQube.
+
+The pages are registered in `src/main/java/org/sonarsource/plugins/example/web/MyPluginPageDefinition.java`, and their respective front-end source code is located in `src/main/js/`. These examples use different stacks to demonstrate different possibilities:
+
+* React JS examples (recommended, SonarQube uses React v16.8):
+  * `src/main/js/portfolio_page/`
+  * `src/main/js/admin_page/`
+* Backbone JS example: `src/main/js/project_page/`
+* Vanilla JS example: `src/main/js/global_page/`
+
+#### Helper APIs exposed by SonarQube
+
+There are several helper APIs exposed by SonarQube, like functions to make authenticated API requests, or parsing measures data. SonarQube also exposes several React components, in case you want to use React for your own pages.
+
+You can find the full list of exposed helpers [here](https://github.com/SonarSource/sonarqube/blob/master/server/sonar-web/src/main/js/app/components/extensions/exposeLibraries.ts).
+
+The included pages contain several examples:
 
-* run "npm install" to setup your env
-* run "npm test" to trigger your tests
-* run "npm start" to start a proxy http server on port 3000 to debug your JS code
+* **API calls (`window.SonarRequest`)**  
+  Check `src/main/js/common/api.js` for some examples.
+* **Measure helpers (`window.SonarMeasures`)**  
+  Check `src/main/js/portfolio_page/components/MeasuresHistory.js` for some examples.
+* **React Components (`window.SonarComponents`)**  
+  Check `src/main/js/portfolio_page/components/MeasuresHistory.js`, `src/main/js/portfolio_page/components/VersionsMeasuresHistoryApp.js` and `src/main/js/admin_page/components/InstanceStatisticsApp.js` for some examples.
+* **Localization (`window.t()` and `window.tp()`)**  
```

Comment:
Documenting localization is ok too I guess

Reply:
Could we do it differently?

Reply:
No I just meant that indeed we can keep documenting SonarRequest and the localization part.